miniapps/nurbs/nurbs_ex1.cpp

```diff
@@ -10,6 +10,9 @@
 //               nurbs_ex1 -m ../../data/disc-nurbs.mesh -o -1
 //               nurbs_ex1 -m ../../data/pipe-nurbs.mesh -o -1
 //               nurbs_ex1 -m ../../data/beam-hex-nurbs.mesh -pm 1 -ps 2
+//               nurbs_ex1 -m ../../data/two-squares-nurbs.mesh -o 1 -rf ../../data/two-squares.ref
+//               nurbs_ex1 -m ../../data/two-squares-nurbs-rot.mesh -o 1 -rf ../../data/two-squares.ref
+//               nurbs_ex1 -m ../../data/two-squares-nurbs-autoedge.mesh -o 1 -rf ../../data/two-squares.ref
```

Comment:
These sample runs are helpful. Since there is code for the 3D case, can you please add a simple 3D mesh and corresponding sample run? This will ensure the 3D case is tested.

Reply:
1. I did not want to flood `mfem/data` with many meshes so I left a 3D case out. I will add a 3D version.

2. The C-mesh is indeed more complex than the sample runs, but the problem is very  similar. The current PR is the most important step towards enabling a C-mesh, but it also requires the other PR's mentioned in the current PR.